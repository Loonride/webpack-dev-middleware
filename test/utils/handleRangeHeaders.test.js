import handleRangeHeaders from '../../src/utils/handleRangeHeaders';

describe('handleRangeHeaders', () => {
  it('should return content in range with valid range header', () => {
    const content = 'abcdef';
    const req = {
      headers: {
        range: 'bytes=1-4',
      },
    };

    const res = {
      setHeader: jest.fn(),
    };

    const contentRes = handleRangeHeaders(content, req, res);
    expect(contentRes).toEqual('bcde');
    expect(res.statusCode).toEqual(206);
    expect(res.setHeader.mock.calls).toMatchSnapshot();
  });

  it('should handle malformed range header', () => {
    const content = 'abcdef';
    const req = {
      headers: {
        range: 'abc',
      },
    };

    const res = {
      setHeader: jest.fn(),
    };

    const contentRes = handleRangeHeaders(content, req, res);
    expect(contentRes).toEqual('abcdef');
    expect(res.statusCode).toBeUndefined();
    expect(res.setHeader.mock.calls).toMatchSnapshot();
  });

  it('should handle unsatisfiable range', () => {
    const content = 'abcdef';
    const req = {
      headers: {
        range: 'bytes=10-20',
      },
    };

    const res = {
      setHeader: jest.fn(),
    };

    const contentRes = handleRangeHeaders(content, req, res);
    expect(contentRes).toEqual('abcdef');
    expect(res.statusCode).toEqual(416);
    expect(res.setHeader.mock.calls).toMatchSnapshot();
  });

  it('should handle multiple ranges', () => {
    const content = 'abcdef';
    const req = {
      headers: {
        range: 'bytes=1-2,4-5',
      },
    };

    const res = {
      setHeader: jest.fn(),
    };

    const contentRes = handleRangeHeaders(content, req, res);
    expect(contentRes).toEqual('abcdef');
    expect(res.statusCode).toBeUndefined();
    expect(res.setHeader.mock.calls).toMatchSnapshot();
  });
});
