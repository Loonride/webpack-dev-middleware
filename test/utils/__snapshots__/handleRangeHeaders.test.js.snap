// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleRangeHeaders should handle malformed range header 1`] = `
Array [
  Array [
    "Accept-Ranges",
    "bytes",
  ],
]
`;

exports[`handleRangeHeaders should handle multiple ranges 1`] = `
Array [
  Array [
    "Accept-Ranges",
    "bytes",
  ],
]
`;

exports[`handleRangeHeaders should handle unsatisfiable range 1`] = `
Array [
  Array [
    "Accept-Ranges",
    "bytes",
  ],
  Array [
    "Content-Range",
    "bytes */6",
  ],
]
`;

exports[`handleRangeHeaders should return content in range with valid range header 1`] = `
Array [
  Array [
    "Accept-Ranges",
    "bytes",
  ],
  Array [
    "Content-Range",
    "bytes 1-4/6",
  ],
]
`;
